terraform/tfschema/time_type.go

```diff
@@ -106,6 +106,10 @@ func (t TimeType) ValueFromTerraform(ctx context.Context, in tftypes.Value) (att
 func (t TimeType) Validate(ctx context.Context, in tftypes.Value, path *tftypes.AttributePath) diag.Diagnostics {
 	var diags diag.Diagnostics
 
+	if in.Type() == nil {
```

Comment:
How does this change relate to the retry changes above?

Reply:
It's related with the second linked issue
It ended up being something simple and triggered when the cache is enabled

I'm fine opening up a PR for this particular change

Reply:
Ah, gotcha.